src/main/java/seedu/address/logic/commands/book/EditBookCommand.java

```diff
@@ -0,0 +1,212 @@
+    }
+
+    /**
+     * Stores the details to edit the book with. Each non-empty field value will replace the
+     * corresponding field value of the book.
+     */
+    public static class EditBookDescriptor {
+        private BookName bookName;
+        private Isbn isbn;
+        private List<Author> authors;
+        private Set<Tag> tags;
+
+        public EditBookDescriptor() {
+        }
+
+        /**
+         * Copy constructor.
+         * A defensive copy of {@code tags} is used internally.
+         */
+        public EditBookDescriptor(EditBookDescriptor toCopy) {
+            setBookName(toCopy.bookName);
+            setIsbn(toCopy.isbn);
+            setAuthors(toCopy.authors);
+            setTags(toCopy.tags);
+        }
+
+        /**
+         * Returns true if at least one field is edited.
+         */
+        public boolean isAnyFieldEdited() {
+            return CollectionUtil.isAnyNonNull(bookName, isbn, authors, tags);
+        }
+
+        public void setBookName(BookName bookname) {
+            this.bookName = bookname;
+        }
+
+        public Optional<BookName> getBookName() {
+            return Optional.ofNullable(bookName);
+        }
+
+        public void setIsbn(Isbn isbn) {
+            this.isbn = isbn;
+        }
+
+        public Optional<Isbn> getIsbn() {
+            return Optional.ofNullable(isbn);
+        }
+
+        public void setAuthors(List<Author> authors) {
+            this.authors = authors;
+        }
+
+        public Optional<List<Author>> getAuthors() {
+            return Optional.ofNullable(authors);
+        }
+
+        /**
+         * Sets {@code tags} to this object's {@code tags}.
+         * A defensive copy of {@code tags} is used internally.
+         */
+        public void setTags(Set<Tag> tags) {
+            this.tags = (tags != null) ? new HashSet<>(tags) : null;
+        }
+
+        /**
+         * Returns an unmodifiable tag set, which throws {@code UnsupportedOperationException}
+         * if modification is attempted.
+         * Returns {@code Optional#empty()} if {@code tags} is null.
+         */
+        public Optional<Set<Tag>> getTags() {
+            return (tags != null) ? Optional.of(Collections.unmodifiableSet(tags)) : Optional.empty();
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            // short circuit if same object
+            if (other == this) {
+                return true;
+            }
+
+            // instanceof handles nulls
+            if (!(other instanceof EditBookDescriptor)) {
+                return false;
+            }
+
+            // state check
+            EditBookDescriptor e = (EditBookDescriptor) other;
+
+            return getBookName().equals(e.getBookName())
+                    && getIsbn().equals(e.getIsbn())
+                    && getAuthors().equals(e.getAuthors())
+                    && getTags().equals(e.getTags());
```

Comment:
Good implementation of the equals method and good use of guard clauses to ensure that the reader is not distracted from the alternative paths.